Controls for CIS sle15

truzzon opened the pull request:
#### Description:

- Move SLE15 CIS Benchmark over to Control File
- Add Checks, to Benchmark that were newly added since the last PR for this Benchmark file.

### ToDos/Need Help:

- [x] Three of the Unit Tests are currently failing:

   I'm running the Test in a Docker container (OpenSUSE15.2)
```
99% tests passed, 3 tests failed out of 370

Label Time Summary:
quick    = 112.95 sec*proc (113 tests)

Total Test time (real) = 400.52 sec

The following tests FAILED:
        138 - validate-ssg-eks-cpe-oval.xml (Failed)
        143 - validate-ssg-eks-oval.xml (Failed)
        145 - validate-ssg-eks-ds-1.2.xml (Failed)
Errors while running CTest
```

Can you help me out?

Best Regards

ggbecker commented:
> Can you help me out?

this problem should have been fixed by: https://github.com/ComplianceAsCode/content/pull/7982

You can simply rebase your branch on latest master and that should do the trick

truzzon commented:
OK, the tests are passing locally and online.

I hope the formatting changes are no inconvenience. I noticed to late, that my "format on save" was enabled.

Ready for Review.

ggbecker commented on products/sle15/profiles/cis.profile:
```diff
@@ -1,970 +1,22 @@
 
 metadata:
-    version: 1.0.0
+    version: 2.0.0
```
```suggestion
    version: 1.0.1
```

The version here is intended to be the same as in the policy. Please change in other cis profiles files as well.

truzzon replied:
Done.

truzzon commented:
> > I hope the formatting changes are no inconvenience. I noticed to late, that my "format on save" was enabled.
> 
> I'm afraid you have to remove all those formatting changes before we can merge this one. Please, try to keep only the relevant changes.

OK, do I have to "rebuild" the branch, so there are clean commits in this PR or is a single commit reverting the formatting changes enough?

ggbecker commented:
> OK, do I have to "rebuild" the branch, so there are clean commits in this PR or is a single commit reverting the formatting changes enough?

It's better if you edit the specific commit. You can do that by running `git rebase HEAD~~ -i` and then `edit` the commit a4e6ae772a697e3bfb50e2c803bc2c1d0ff98118. This way you can do all the changes, then run `git ci -a --amend` to rewrite the commit with the new changes and then `git rebase --continue`, at last you force push.

truzzon commented:
Thanks, today I learned :) I never had to do it before. Sorry to bother you with this kind of questions.

ggbecker commented:
I'm assuming the controls file is just a copy from the SLE12, do you plan to check the references if they actually exist in SLE15 CIS benchmark, I've seen that some ids don't exist in the SLE15 already, there might be a lot of them.

truzzon commented:
> I'm assuming the controls file is just a copy from the SLE12, do you plan to check the references if they actually exist in SLE15 CIS benchmark? I've seen that some ids don't exist in the SLE15 already, there might be a lot of them.

I've already checked the references, removed the stuff that doesn't exist in the in the sle 15 benchmark. I also added and updated the references, for checks that were added since my past PRs.

ggbecker commented:
> > I'm assuming the controls file is just a copy from the SLE12, do you plan to check the references if they actually exist in SLE15 CIS benchmark? I've seen that some ids don't exist in the SLE15 already, there might be a lot of them.
> 
> I've already checked the references, removed the stuff that doesn't exist in the in the sle 15 benchmark. I also added and updated the references, for checks that were added since my past PRs.
> 
> Plan is to do the same with cis_sle12 in the next few days.

I think I hit the sweet spot then, because the `- id: 2.2.1.4` doesn't exist in CIS_SUSE_Linux_Enterprise_15_Benchmark_v1.1.0.pdf, so this one should probably go out.

truzzon commented:
> > > I'm assuming the controls file is just a copy from the SLE12, do you plan to check the references if they actually exist in SLE15 CIS benchmark? I've seen that some ids don't exist in the SLE15 already, there might be a lot of them.
> > 
> > I've already checked the references, removed the stuff that doesn't exist in the in the sle 15 benchmark. I also added and updated the references, for checks that were added since my past PRs.
> > 
> > Plan is to do the same with cis_sle12 in the next few days.
> 
> I think I hit the sweet spot then, because the `- id: 2.2.1.4` doesn't exist in CIS_SUSE_Linux_Enterprise_15_Benchmark_v1.1.0.pdf, so this one should probably go out.

Ok, then I have to check the numbers again. Maybe the chapter was moved, but the headline was the same.

truzzon commented:
I found some references, that I had to update. Thanks for pointing it out.
I'm currently running the tests and push it, when everything was successful.
Should I edit the commits again or can I commit it right away?

On thing, I noticed: (linux_os/guide/system/network/network-wireless/wireless_software/wireless_disable_interfaces/rule.yml)
Why is this causing a build to fail?
```
(...)
references:
    cis-csc: 11,12,14,15,3,8,9
    cis@rhel7: 3.1.2
    cis@rhel8: 3.5
    cis@sle12: 3.1.2
    cis@sle15: 3.1.2
(...)
```
While this is ok?
```
(...)
references:
    cis-csc: 11,12,14,15,3,8,9
    cis@rhel7: 3.1.2
    cis@rhel8: "3.5" 
    cis@sle12: 3.1.2
    cis@sle15: 3.1.2
(...)
```

This happens on multiple occasions...

ggbecker commented:
> Why is this causing a build to fail?

It's because the build system (python) interprets `3.5` as a floating value. That's why we need to use quotes to explicitly state that this is a string instead of a number. The build system code probably needs some improvements in this regards to always interpret it as a string.

Mab879 commented:
/ok-to-test

ggbecker commented on products/sle15/profiles/cis.profile:
```diff
@@ -1,970 +1,22 @@
-    SUSE Linux Enterprise 15 Benchmark™, v1.0.0, released 06-30-2020.
+    This profile defines a baseline that aligns to the "Level 2 - Server"
+    configuration from the Center for Internet Security®
+    SUSE Linux Enterprise 15 Benchmark™, v1.0.1, released 09-17-2021.
```
```suggestion
    SUSE Linux Enterprise 15 Benchmark™, v1.1.0, released 09-17-2021.
```

ggbecker commented on products/sle15/profiles/cis_workstation_l1.profile:
```diff
@@ -0,0 +1,22 @@
+description: |-
+    This profile defines a baseline that aligns to the "Level 1 - Workstation"
+    configuration from the Center for Internet Security®
+    SUSE Linux Enterprise 15 Benchmark™, v1.0.1, released 09-17-2021.
```
```suggestion
    SUSE Linux Enterprise 15 Benchmark™, v1.1.0, released 09-17-2021.
```

ggbecker commented on products/sle15/profiles/cis_server_l1.profile:
```diff
@@ -0,0 +1,22 @@
+description: |-
+    This profile defines a baseline that aligns to the "Level 1 - Server"
+    configuration from the Center for Internet Security®
+    SUSE Linux Enterprise 15 Benchmark™, v1.0.1, released 09-17-2021.
```
```suggestion
    SUSE Linux Enterprise 15 Benchmark™, v1.1.0, released 09-17-2021.
```

ggbecker commented on products/sle15/profiles/cis_workstation_l2.profile:
```diff
@@ -0,0 +1,22 @@
+description: |-
+    This profile defines a baseline that aligns to the "Level 2 - Workstation"
+    configuration from the Center for Internet Security®
+    SUSE Linux Enterprise 15 Benchmark™, v1.0.1, released 09-17-2021.
```
```suggestion
    SUSE Linux Enterprise 15 Benchmark™, v1.1.0, released 09-17-2021.
```

ggbecker reviewed (changes requested):
I believe the version in the description is still wrong. Please double check that.

truzzon commented:
> I believe the version in the description is still wrong. Please double check that.

I have corrected the commit and force-pushed it again. Is that the problem with the failing tests? I didn't change anything else.

JAORMX commented:
/retest

truzzon commented:
I hope, I'm not stating the obvious, but the test failed, because something was/is wrong with the container registry according to the logs.

```
error: build error: failed to pull image: After retrying 2 times, Pull image still failed due to error: Error determining manifest MIME type for docker://registry.fedoraproject.org/fedora-minimal:35: Error reading manifest sha256:3624a07969e7e7f078f1b399ce7c565b2518aa50fde80128b7627f7ca31b9688 in registry.fedoraproject.org/fedora-minimal: manifest unknown: manifest unknown
```

ggbecker commented on linux_os/guide/system/permissions/files/permissions_important_account_files/file_permissions_backup_etc_shadow/rule.yml:
```diff
@@ -45,6 +45,8 @@ template:
         filemode@debian9: '0640'
         filemode@debian10: '0640'
         filemode@debian11: '0640'
+        filemode@sle12: '640'
+        filemode@sle15: '640'
```
Is there any particular reason why not to have the left zero here as other distributions? The rest of the rule will render descriptions and etc with `0640` anyway.

truzzon replied:
6.1.6 in the sle15 benchmark recommends 640 or more restrictive. So I guess both would be fine. If you want to have it on 000 for consistency , I'll update it. And tbh, I've barely seen a system where this file even existed.

truzzon replied:
I have removed the changes. Better safe (000) than sorry (640).

truzzon replied:
I would leave it the way it is, to keep it uniform and because both 640 and 000 are correct implementations of 6.1.6.

If it proofs to be too restrictive, I or someone else will update the value later.
